### ISSUE TITLE
Error with Social plugin when custom scheme used

### ISSUE THREAD

**Issue by zhyatt** [squidfunk/mkdocs-material#2965]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [ ] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

An error is thrown when the social plugin is enabled and a custom scheme is used for the palette.

### Expected behaviour

Proper build occurs with social plugin enabled while using a custom scheme. When using `primary` instead of `scheme` in configuration the build works properly:
```
    palette:
      - primary: indigo
```

### Actual behaviour

Following error is thrown:
```
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/__main__.py", line 183, in build_command
    build.build(config.load_config(**kwargs), dirty=not clean)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/build.py", line 249, in build
    config = config['plugins'].run_event('config', config)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/plugins.py", line 94, in run_event
    result = method(item, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/material/plugins/social/plugin.py", line 70, in on_config
    if palette["primary"]:
KeyError: 'primary'
```

### Steps to reproduce

1. Install MkDocs and Material Insiders via pip
2. Add `docs/index.md` file with any content
3. Add `stylesheets/extras.css` with content such as:
```
[data-md-color-scheme="nano"] {
  --md-primary-fg-color:        #4a90e2;
  --md-primary-fg-color--light: #F4FAFF;
  --md-primary-fg-color--dark:  #000034;

  --md-footer-bg-color:         #000034;

}
```
4. Add configuration as noted below
5. Run `mkdocs build`

### Package versions

- Python: `Python 3.9.6`
- MkDocs: `mkdocs, version 1.2.2`
- Material: `Version: 7.2.4+insiders.2.13.1`

### Configuration

```yaml
site_name: Site Name
theme:
  name: 'material'
  palette:
      - scheme: nano
plugins:
    - social
extra_css:
  - 'stylesheets/extras.css'
```

### System information

- Operating system: macOS Big Sur 11.0.1
- Browser: Brave, Version 1.27.109 Chromium: 92.0.4515.115 (Official Build) (x86_64)

**Comment by zhyatt** [comment]
Discovered when validating fix for #2948

**Comment by marens101** [comment]
Can repro
mkdocs: 1.2.2
mkdocs-material: 7.2.4+insiders.2.13.1

**Comment by squidfunk** [comment]
The error is fixed in the latest commit on the Insiders repository.

**Comment by squidfunk** [comment]
Released as 7.2.5-insiders-2.13.2.
